### PR TITLE
fix: respected style field from package.json

### LINKED ISSUE #683: Style field from package.json not respected
**What is the current behavior?**

When @import-ing a package (blueprintjs/core) from a CSS file, the `"style"` field from the package's package.json file is not used. So JS files are read and this error occurs:
```js
VM71158:1 Uncaught Error: Module build failed: ModuleBuildError: Module build failed: Unknown word (6:1)

  4 |  * Licensed under the terms of the LICENSE file distributed with this project.
  5 |  */
> 6 | export * from "./accessibility";
    | ^
  7 | export * from "./common";
  8 | export * from "./components";
  9 | 

    at runLoaders ...
```

**If the current behavior is a bug, please provide the steps to reproduce.**

I've uploaded a full reproduction at https://github.com/shravan2x/cssLoader-bug-styleUnrecgonized. Just do 
1. npm install
2. npm start
3. The result files will be in `dist`. My repro does not include a webserver in the interest of keeping it simple, but just opening the app.[hash].js file in a text editor will make the error clear.

**What is the expected behavior?**

The "style" field in the package.json file is read and the respective CSS file is imported.

NPM version: 5.6
Node.JS version: 8.8.1

### PR DESCRIPTION
**Pull request by cap-Bernardito**
This PR contains a:

- [x] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Fixes #683 
- respected style field from package.json
- add restrictions

### Breaking Changes

No

### Additional Info

No

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -47,9 +48,19 @@ export default function loader(content, map, meta) {
   plugins.push(icssParser({ urlHandler }));
 
   if (options.import !== false && exportType === 'full') {
+    const resolver = this.getResolve({
+      mainFields: ['css', 'style', 'main', '...'],
+      mainFiles: ['index', '...'],
+      extensions: ['.css'],
+      restrictions: [/\.css$/i],
+    });
```
Comment (evilebottnawi) [inline src/index.js:56]: conditionNames: ['style'],

**Comment by arshaw** [comment]
@cap-Bernardito, question for you: did this ever work? Was the `style` field in package.json ever respected? [This old issue here](https://github.com/webpack-contrib/css-loader/issues/51) makes me think that it was.

**Comment by evilebottnawi** [comment]
@arshaw And yes and no, before this commit you need to setup the `style` field in your configuration, now you don't need this, because it is out of box

File: src/index.js [hunk 2]
```diff
@@ -47,9 +47,20 @@ export default function loader(content, map, meta) {
   plugins.push(icssParser({ urlHandler }));
 
   if (options.import !== false && exportType === 'full') {
+    const resolver = this.getResolve({
+      mainFields: ['css', 'style', 'main', '...'],
+      mainFiles: ['index', '...'],
+      extensions: ['.css'],
+      restrictions: [/\.css$/i],
```
Comment (andersk) [inline src/index.js:54]: This restriction broke `@import` for files with an extension other than `.css` (#1164). Can we remove it (#1165)?

**Pull request merged**